### ISSUE TITLE
[Bug]: Multiple rescans running paralelly

### ISSUE THREAD

**Issue by gautamkrishnar** [tubearchivist/tubearchivist#624]
### I've read the documentation

- [X] I'm running the latest version of Tube Archivist and have read the [release notes](https://github.com/tubearchivist/tubearchivist/releases/latest).
- [X] I have read the [how to open an issue](https://github.com/tubearchivist/tubearchivist/blob/master/CONTRIBUTING.md#how-to-open-an-issue) guide, particularly the [bug report](https://github.com/tubearchivist/tubearchivist/blob/master/CONTRIBUTING.md#bug-report) section.

### Operating System

Linux Arm

## Describe the bug
I am running tube archivist with a hourly rescan interval. Sometimes, the rescan gets stuck and will go into endless scanning as shown in the image. 

<img width="1185" alt="Screenshot 2023-12-06 at 1 39 05 AM" src="https://github.com/tubearchivist/tubearchivist/assets/8397274/20d85e1a-652a-447e-982a-15b14e7b231a">

## Steps To Reproduce
1. Set hourly rescan
2. wait

## Expected behavior
When re-scanning, Tubearchivist should cancel the previously running scan. there should be an option in the UI to enable this as a checkbox so that this will be an opt-in feature. 

### Relevant log output

```shell
n/a
```

### Anything else?

n/a

**Comment by bbilly1** [comment]
That shouldn't be happening, you should only be able to have one rescan process running:
https://github.com/tubearchivist/tubearchivist/blob/master/tubearchivist/home/tasks.py#L178-L182

There must be something else going on. What do the logs say? And also probably rescan once per hour is too much, you might be getting blocked by YT at this point.

Or maybe that's just the notification getting stuck?

**Comment by lue30499** [comment]
Hello @bbilly1 I think have the same or similar thing happening...

It gets stuck like this.

<img width="1190" alt="Screenshot 2023-12-10 at 07 54 41" src="https://github.com/tubearchivist/tubearchivist/assets/63104612/e1124f69-364a-4f9d-b831-6b261cd490b9">

TA container Logs:
[ta_logs.txt](https://github.com/tubearchivist/tubearchivist/files/13625638/ta_logs.txt)

Regards

**Comment by gautamkrishnar** [comment]
For me its just youtube throttling my IP

**Comment by lue30499** [comment]
Ah, I see, I think that makes my issue different then. Since yours is backing up with scanning subscriptions, mine is reindexes... 

I'll make another issue, if I see it again, as I restarted TA.
